recipe/build.sh

```diff
@@ -3,54 +3,101 @@
 # unset the SUBDIR variable since it changes the behavior of make here
 unset SUBDIR
 
-USE_NONFREE=no
-
 declare -a _CONFIG_OPTS=()
 
-# We do not care what the defaults are as they could change. Be explicit
-# about every flag.
-if [[ ${USE_NONFREE} == yes ]]; then
-  _CONFIG_OPTS+=("--enable-nonfree")
-  _CONFIG_OPTS+=("--disable-gpl")
-  _CONFIG_OPTS+=("--disable-gnutls")
-  _CONFIG_OPTS+=("--enable-openssl")
-  # The Cisco GPL-compliant wrapper (you need to get your own binaries for this)
-  _CONFIG_OPTS+=("--enable-libopenh264")
-  # GPL-3.0
-  _CONFIG_OPTS+=("--enable-libx264")
+if [[ ${target_platform} == win-64 ]]
+then
+  # the target-os and toolchain picked up by default is mingw, so we have to configure
+  # these flags ourselves to get it to build properly
+  _CONFIG_OPTS+=("--ld=${LD}")
+  _CONFIG_OPTS+=("--target-os=win64")
+  _CONFIG_OPTS+=("--toolchain=msvc")
+  _CONFIG_OPTS+=("--host-cc=${CC}")
+  _CONFIG_OPTS+=("--enable-cross-compile")
+  # ffmpeg by default attempts to link to libm
+  # but that doesn't exist for windows
+  _CONFIG_OPTS+=("--host-extralibs=")
+  # we don't want pthreads on win, it's not posix compliant
+  _CONFIG_OPTS+=("--disable-pthreads")
+  _CONFIG_OPTS+=("--enable-w32threads")
+  # manually include the runtime libs
+  _CONFIG_OPTS+=("--extra-libs=ucrt.lib vcruntime.lib oldnames.lib")
+  _CONFIG_OPTS+=("--disable-stripping")
+  export PKG_CONFIG_PATH=${PREFIX}/lib/pkgconfig:${PKG_CONFIG_PATH}
+  # unistd.h is included in ${PREFIX}/include/zconf.h
+  if [[ ! -f "${PREFIX}/include/unistd.h" ]]; then
+      UNISTD_CREATED=1
+      touch "${PREFIX}/include/unistd.h"
+  fi
+
 else
-  _CONFIG_OPTS+=("--disable-nonfree")
-  _CONFIG_OPTS+=("--enable-gpl")
-  _CONFIG_OPTS+=("--enable-gnutls")
-  # OpenSSL 3 will be Apache-licensed so we can revisit this later:
-  # https://github.com/openssl/openssl/commit/151333164ece49fdba3fe5c4bbdc3333cd9ae66d
-  _CONFIG_OPTS+=("--disable-openssl")
-  # The Cisco GPL-compliant wrapper (you need to get your own binaries for this)
+  # we choose libopenh264 instead of x264 to make this LGPL
+  # we don't have these packages for win
   _CONFIG_OPTS+=("--enable-libopenh264")
-  # GPL-3.0
-  _CONFIG_OPTS+=("--enable-libx264")
+  _CONFIG_OPTS+=("--enable-libopus")
+  _CONFIG_OPTS+=("--enable-libmp3lame")
+  # Our win packages don't have .pc files for these
+  _CONFIG_OPTS+=("--enable-libopenjpeg")
+  _CONFIG_OPTS+=("--enable-libvorbis")
+  # we don't want pthreads on win, it's not posix compliant
```

Comment:
This is not accurate. Copy/paste mistake?

Reply:
windows isn't posix compliant, right?

Reply:
i'll clarify

Reply:
This is the "no windows" block, we wo do need pthreads here.

Reply:
sorry can you clarify? I changed this to "pthreads doesn't exist on windows (without relying on msys2, etc)". Is that ok?

Reply:
THis comment is on the "not windows" block (we are in the else block).

Reply:
oh so it is.

Reply:
oh so it is. removed